### ISSUE TITLE
Rendering fails with SVG bitmap image outside canvas

### ISSUE THREAD

**Issue by lostminds** [thorvg/thorvg#1951]
Another issue encountered exploring embedded bitmap data in SVGs. It seems that if you construct an svg where a bitmap-based element (like a symbol referencing an image with embedded png data) placed entirely outside the svg canvas, rendering of the entire svg will fail.

Doing this from within Godot (4.2.1) will give you an error `ImageLoaderSVG: Couldn't draw ThorVG pictures on canvas.` which will cause load_svg_from_string() to fail. And the same svg will just not render in the web viewer https://thorvg.github.io/thorvg.viewer/ (so I think the same issue is present in the latest ThorVG version).

Here's an example file you can try:

![RepeatedGodot_Fails](https://github.com/thorvg/thorvg/assets/17763524/dbecd60d-6dd6-4698-8033-09f3d04ef4e0)

It contains the embedded Godot icon png data repeated in three elements, where two fit in the area and the third is completely outside it. Trying to show this in https://thorvg.github.io/thorvg.viewer/ just fails silently for me.

Below is a modified version of the same file, with the only change that the image symbol instance outside the art board is removed, and this does not fails (just with the #1949 rendering issue).  

![RepeatedGodot_OK](https://github.com/thorvg/thorvg/assets/17763524/36deef9c-81d4-490a-98a2-9c90fc928858)

[RepeatedGodotTestSVGs.zip](https://github.com/thorvg/thorvg/files/14128367/RepeatedGodotTestSVGs.zip)

**Comment by capnm** [comment]
I took a quick look and simplified the demo example.
```xml
<svg xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" 
    viewBox="0 0 300 128"
>

<defs>
<symbol id="imageName">
<image width='100' height='100'
xlink:href="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAIAAAACACAYAAADDPmHLAAAAAXNSR0IArs4c6QAAFgdJREFUeJztnXlsXMd5wH/zrr14k5KokyIlUpTkS9Zh+T7j2LEbu21cO0iB1HYcJEWLAImdBEkdOy4KGEibC20RIIeKpD5iJ2kcB0ks14lj19FlRdZhSZRMiRR1UzyXe75j+scuqd0lRe6Sb5e70v6AxXLfm5n3PX7fmzfzzTczQkpJS9s6BAIATUhSsZPfKipAsymtjwMbgabkp5oys8kQ0J38bNWF9gJw1E5qTs1IbMmEniWSI4feRTS3rh07OZERJA3gPkfKL5NQfJniZ6sixLPAK6kGkKp8ACEkSmqu0ROjCYF2R/KaI+UvKSu/lNjoSPlLR/Ia0A4TKx9AGX3qR0kxgodMKXeBvLNQUpdxG3mnKeUuS4qHYLzyBaAl/hBjJwEhkV8H+WRmcUIIdMOH1+tHMzwoqopIr0TKFBiJg2PbWPEY0WgYMx5ByrR2nFfivABiFfCUEImTo4+9OqdhPg6ClJrgmYmU7/H6qa6dh89fiaYbKIpKZu1RpvAIBIqioukGXl8Ar68Cx7GwLTMz6c0ioa43R7WmCQW1vn4BimDUCB4C+d20CwiFqtoGAhW1CKX8tBc7QlHweANouk48FgXSaoNbFMFB4H1NJHQ5plFNyHaJsyk1tapq1NQ34vEECiB6GTfxeALU1DeiqlracUeySRPKitHfig2YEkzJdwDv6InEkz8XTTMKJnQZd9E0g6qauQiRVnP74o7znbjjYJOsARTBfZmt/cqa+rLyLwI03aCypj7z8IdVRfkogOJISdLJM4bH6y9X+xcRHk8Aj9efdsx2nC/bjoOiC62ZFCePEIKKyrpCy1gmz1RU1iFEWq/tWo9qLFWSvv0xdMOHktFwKFP6KKqGYXjTjsUd8yGFDBevN6OqKHPx4PFVpP2WUl6rkBjRG0MzPIWUqUwB0TQ9/YCgaZwBKGrmAGKZiwVFy3i1y4QBpI3nl337Fy8T6LamrO1LnLIBXOKUDeAS56Lq8Cs+P/qcJrS5S8AyCb33e1fKDVx+I8JbgXX2GObZY9iRoCvlFgMlbQBCUdDqFqDPa0KfuwS1qiFt8DN+7CBm/8kZXUOrbsBovhKEQF+wHB/gBPuxeo9hnunGPHcC6dhTllOslKQBqL4KjCWr8TStQqQ4N2RGOt9l12O+9fKMruVbfQOku1BRKuswKuswWq6CWJjosQPEu/Zhh4dndK3ZoGQMQAiBNmcxnubL0ec1j1PKRKi1jXgWtRE7fmha1zQal6LNWTx5Io8fb+tafK1XY545Rqx7H+aZLqTjTOuahaZkDKDqtk+gVNTmnM+78jriJztzrqaFouBbdX3W6SUCbV4T2rwmZDTI0Os/KYlXQ8n0ApRpjlEo/kqMxSumTpiB0diCMs1RUeGpKAnlQwkZgDTHBTlmjbd1bc7xjJ62ddO+Hvb0ZS00pWMAM/inKoEajMaWrNPr85pQq+dM+3qOFZ923kJTMm0Aac7sn+ppW0fs5AcIVUUYHoTuQ4yOj5tRnFgUaUaRto13Jk8/wPiQ7KKldAxghk+VWj2H2ns/C1MFu9jW1GmmYKayFpKSeQU4sfDMC8lGsS5EQ8noyIzLKBQlYwCyhNyvdrhsAK7jREOzLULWyFjpyFo6BhApnaeqlGQtGQMokx+KvhegqBqeFRvwLV8zbrCnWPGv+RBazVzCB7Yii7xLWBAD8CxZiTPcjzl4Jqd8esMiAlfdigjUlIzyARACo+UqtMYWInveJH6mO6fsWnUDWm0j0a59eRIw5Vr5voBn8Qr8a+4AwDx9hOjBrVhDfZPmEaqG//Kb8DStLi3FZ6D4qwhs/Cj68Q4ie97EmcKZpVbW4VuxAWNh6+haHkS79uZVxrwagN6wiMCa28eUqDe2YDQ2Ez9xmPD+d3Am6C6pvkoC19yDWj2npJWfirFoBVr1HEa2/Ro7NDTuvOLz42+/Dn1xOwgxdt++K27GiQSJn+nKm2xqbf38p1MPBCpqXClYq6yl8vr7Qc2YjIBArarH23QZOBb20FlILmmi1y2g4vq/RAlcfCvPCY8Pz+J2nMEzY4EjQgi8zZdRsf5e1LrG8TEOQmA0tmD1duNEXXCEAeGRwbTfeTEA1eun4vq/Qkw2w1hR0ec2Ycxrwh46iz5vKRUb7oKLeUq6qmEsagcrhrQsKq75CMbSy2GyyTiKijG/GfPEB664mDMNQDS3Xp1W085pXDqjCwhFpfLGv0atmZd9JimzivC5qMjxnp3hPobffnnGvYre011pv133A/hWrM9N+XDpKR9yvmelqh7/qutcF8NVA9Bq5+FpneFQapkL4mm+Ar1hkatlumYAQtUIXH3Hpfk0FwgJ+NfcgaK7105yzQB87degVJRXFsk3ir8yEaruVnluFKLXLcC7fI0bRZXJAqNpNfrcJa6U5Y4BzG9GllcNLSjG/OxjHCfDFQOIdmwvqYCNUkdGg0QObHGlLFcMwLFMwrv/4EZRZaZAAOE//y9OPOZKea41AuNnuon3HHCruDIXINr5HvHe466V56ofILL3bWQJRcOUGs5IP1GXqv5RXDUAx4wR2vna2OBOGfcQ0iG0czOObblaruuuYLPvJNGOHW4Xe8kTObgda7DX9XLzEhMYPbwDu29mCzOUOY89cJroBzvzUnZeDEA6DqH33shH0Zck4b1v5W29gbxFBEkXuilzKj1sXNbAwloflV6dYNTk5ECEbUf6ODMcdUFK92io9HBNSz2L6/xUenVGYhYnB8JsO9LH6aGZySpNd7p8E1GUUcFL6v08cuMyNi4bt849AH9/eyvbj/bxwz8eobtvdidhLKr18chNy7huecOE5z97Wys7u/r5wVudHO0tvgkjRWcAt7TP5Qt3taOrk7+dNjTXs2ZJHd/efJA39ucWbewWN7TN4Yt3r8TQJpd17dI6rlpSy3de72DzvtMFki47isoAbmqbw5fvWZV1el0VPHH3ShwJfzhQWCO4dlkD//QXq7NOryqCz3+4HSnh9feLxwiKxgAW1Ph4/O6V447v2bOHX73yKh0dHZw7d46WlhZWr17FJ/72E1RXJ4JHP//hdg6fCXK8353AyamYW+XhS/eclzU0EuK5555j7959dHZ2UldXS1tbG/fcew9r165Ny/u5O1dw6PQw3X2FkXUq8hYVLFQdb+vVWaf/hzvaWDb3/JJvju3wve99j29989scPXqUYDCIbducO3eO/fsPsPm1zSxdupRFixaiKoL6Cg9vdZzN+nqGpnDFohoW1PjoD8WxneydV5+5tZUVjVUA7Nr1Ho8//gTbt++gt7cXy7IIBkfo6urm9c2v098/wLp1a1GTgZ+KEMyt8vLmwexljR3dg4y70+jNDAotihqgyqdzY9vctGMvvfQSL7/8swvmGRgY4OmnnuYHP/w+Cxcu5PrWBmr9BgPhqSNnF9b6ePaBq5hTmdgboT8U5ys/203XuakbaX5D5baViZjH3t5evvbk1wiFLpzv1Vdfpbq6mkcefXjs2DUt9dRXGPSNzP5CEkUxOXTt0jqUlHCCEydOsGnTf02ZLxaL8a/f+Lex3+ubs4tIeuzm5WPKB6gLGHzm1uVZ5V3TVIeaFPab3/zWpMof5fnnn6ez80jasfXNE/dwCk1RGMCCmvS9bLZt246Z5apgu3fvZmQkMQA1v8aXVZ62xsoJjlVllXdU1lgsxo7t2bm8Hcdh65b0QZwFWcqab4rCACo86bOHDh3KbWXPw4cOA4lXSTYoEwSuqkp2EU0V3sQ1Ojs7cXLwznV0pN9TtrLmm6IwgKFI+tNuW7mNeJnJyRKDWbz/Z8pwUlZrmjKOUghZs6EoDKAno/vWtiK3lT1XJNNnlpMPepLdt9bW1sx9+CalPeOeCiFrNhSFAfy5qx/TPt8NW7duLUqWK3u2tbVRXV2NIyXvHu3Pl4hj7O4ZIGY5+Hw+rrji8qzzbdiwYexvR8KOAsiaDUVhABHTZvO+U2O/m5ubeeihB6fMp6oqTzzxOACb951mJOZusMRExCyH3+xODHV//gtfwDCmnqRx//33s3LVecfRG/tPj71KZpuiMACA//5TV9o/5ZN/90luv+P2C6b3er188UtPsGz5MoJRi5/86WghxATghW3dDIZNFi9exFe/+hUCgQvPgr7xxhv49KcfG/sdiln8+J3CyToVReMJjJo2B08Nc9uqRhQhUFWVm266kZZlyxgaGiI4HMQ0TebPn8/GjRt5+utPceWVV+JIydO/3JfTSNvH1i/Bq6dPybYdyYvbslvKJWY5vH9iiNtXNdLc3MSdd36IYDBIJBIlGAzi9/tZvXo1j37qUR559BE0PeFvcyQ888o+PjibW9xkPj2Brk8PH0UxfFTf/amc8122sJon77uM6oxukpSSSCSC339+2fhg1OKfX9nHnuODmcVMygufvY5af3rVHTVt7v/u2zmV0z6/iqfuvyytrHA4jM/nG9dAHIlZ/Muv3mfXsYGcrgEw/MZPsEdyu8cLkffp4TNl34khHtu0nf/ZeZyoeX7NfSHEmPJjlsMru07wqR9ty1n5cL4ln8p0BmcOnhrm05u287MdPUSSsvr9/jTlxyyHX793ksd+tH1ays83RVcDpDI6YLOw1ke132A4YnJiIMzunkHi1vRDpFYuqOLZB67CkxzHN22HJ3+xl/dmoCBdVbh8UTWL6vzUJGU9ORAZ6zXMhHzWAEVtAPmkLmCwcVkDigLbOvvoDeYv7Gqm5NMAimI0cDboD8X5zZ5y5HLRtQHKFJayAVzilA3gEidvbQC/R+VvVgYYiUuCcYeQKRmJO4wkv0NxiV2eQwgkhqIDuiBgCCp1JfFtCAK6SoUOv9iicy5Pc27zZgCaovDIR9ZPmiYYtegPxRkMx+kPxRkY/YTjHO0doTNHj1mx0twQoGVuBbUBg1q/QV3AoCaQ+K71G1PGBvz2uR/nTbZZ7QVUejUqvRpN9RNvCvn9P3by83d7CiyVu9xz5QL+8Y622RbjguTNAExb8sbe48yrq6TWb1AbMPAbkyyJOgG3ts+btgEEPBoP39DC8nkVUyeehK5zIX709pFpj97d2p7jopkkBowGwiaD4Ti9g2HC0RKcGhaJW3xj8wdpx3RVocavUxswqPEZ1AR0avyJarAmYFDj11lSF6C+IuFbD3inL97Hr2ni3qsWzOgeIOHvt2zJv78xvQ2oU++hPxTnWF+IgbDJUPj8624wZDIYSfweDJuYdrrncGggf7uSF/QVYNoOvcHYpF631CqzwjN98VYvdG/F8ZmUlXoPP3+3p+heaUXnCUytagOe3F4ZqTSmRBrv6Rnk8JncVjFragiwbmkizHx+RtRyLgRSDCBYJEEgqeTRAKbXxUs1AEUIPJqS82CKR1PShmhf2XWcdw6fy6mMNUtqxwzAq6tU+/RxwatTIQBfSrsnGJ1exJLMY3c5b44gacWntVbQcDT9n1wxjXZAY3V6zP10AjB7BtLzNFbnXgv4PVra8pmZ95YN0orndSv6/BmA40A8knO+YCT9KZlf45tyqngmqVW2I+HUYO5ynAvG0mqebCedjKKrYtyrIzgNA8CNLXMnIa9tAGu4D23OxH38C5H5T/rXBxNrEEdNm6GISTBiMRSJMxy1GI6Y4z9Rk+Vzz8/8OT0USYs4zoXj/eGxCavL5lZyrC9MlU9PfjSqfTqVXj3x7Ut8j5736ePbL5nGnQ3W8OQbbM2UvBqAeaYbbc7inPLELIeoaY+L2fPqKl5dZV52M7jGmEn8/fGB8wbwwPrFPLA+t3tJxZHTqwGsHLecy5W8DgaZp49Mqx3w4rZuHJcaPjNZM8Ct9QYk8NL2Y1g5TEEHwHHyumMY5LkGsENDmCcOoy/KzRX64rZj/HbPKeoCBhVejQqvnnQbJ74rPDoVSTdy6rmAR0ubZWzakt/PYOWQNw+e5WPrl4yFjkFCmSNRi2DUPP8ds8b+Dkat9PMxc8zBkyvx7vfzvml23v0AkY7tGAuXIUVuffqhiDntblelN2EgpwYjhOPTb0H39Id58D/fYWGtj1DMIhi1CMeswuxnaFtED+dnbcBU8m4A9sgA4fe34LvMvV0uLoQEwnGbcNzmjEve06hpz8qoZGTvW9gFWIK/IAEh0c5dWKeLZzZMsWMe7yDa/X5BrlWwiKCRd3+H7eIy5xcr5ukjhHYVbpXVghmAtC2C217F6j1WqEuWHObJDwjt+F1ePX+ZFDQmUNoWI1tfJXp4Z3mHoVSkJHpgCyM7fltQ5cMsjAZKxyGy/0/Y/SfxX3kLwjt+vZ5LCSc0SPi9P2Cem53X46wNB8dPd2H1PY93xQa8LVcixSUWoOw4RD/YSfTQu0iXN4HIhVmNB3DMOOF9/0fs6F68beswFq+86HceFdIm1r2f6KGdBenmTUVRBITYoSFCu94geuhdjKZVeBavuOheDTI8TKznALFj+3HCxRPtXBQGMIodGiKyfwvRA1vRGxZiLGpHX7gc1OJYUi1nzDjxk4eJ9xzELNIdVIrKAEaRUhLvPU689zhi7x8x5regz2tGrZ+P4p1ZlG++kZEgVt9J4me6sE4dcX2TJ7cpSgNIRVomsZ4OYj0dAKiBKrS6BWj189HqFqBW1hXGNz8BAok13Ifddwqr/yRm/8miqt6zoegNIBM7NIwdGibWcxAAYXhRq+pR/VUo/iqUQBVqoBrFV43i9c/YOATgREPY4SFk8tpOeAgnPIwd7HNtB8/ZouQMIBMZj2KdO4HFiXHnhKqh+CoQmo5QdYSmg6ol/9ZANRLxCraFtONIywLbRFoW0jaRtokTGZnVblq+KXkDmAxpW66trHGxcol5X8pkUjaASxwFGEo9IMnPBoVlZp8JdDukAGlhp45d2NGoMoXDyVziXtA1zgCsEu/WlLkwVsaeBUi6FWBr6rFotDjWsS/jPrFIupNKCLFF0YX2QupBMx4pevdlmdxxbIt4xoLThqK/qABHSakFpJSMBItjM4My7jES7M+cZbwF6FJsbBQhnk09E4uGicWKb6PjMtMjFgsRy3i1q4ryrCWtMT/AKyA2pyYIDvZhWcWxsVGZ6WOZcYKD4yaYvgb8CkBRARXQBZ8Dxl4SUjoMD5zFMstGUKpYZpzhwbNImdb/jxiK8rlRvad6Ag8KlIdTU9q2xWD/6fLroASJxUIM9p/GTm/QS0XwMNAxekCtr0+spGVJAbCPhGHcnJKHWDSMbcXRdQ8iy928yswOjm0RHD5HeGSIzGV6hOAZAf/hIMc2z1Tr6xeMKh+ZyPBm8twtqZltyyQaCWKZMRACAQhFIMoR/rOKxMGxTMx4lHBwgJHgwHiHD0gheAb4OiRjHJJGIFpa1yULSliLEMlveNCRbAKKY5PbMtMloggelvDT1IOjj60CEyof4KeGoqwh0WIsU5q8ltThTzPr6dGXg3IB5aMlJmp0AHepinIfCcdBmdJgS1JndwEdSV2Oe1lLQEgpWbZibVoCLWWWzujYoO04eFRjadwxH5JSXougCUkT4M5Gg2WmyxCCLiTdQogthqK/GLPjXWqysZ66LIeV7A6OPv2dHTv5f1g7iGD/R7z4AAAAAElFTkSuQmCC"/>
</symbol>
</defs>

<g id="Images">
    <!--use id="Image 1" x="299" y="10" href="#imageName"/-->
    <use id="Image 2" x="300" y="10" href="#imageName"/>
</g>

</svg>
```
 
Theoretically at least, trying to paint outside the viewport looks wrong. Inside Godot you get from draw() an `InsufficientCondition` error.

![Screenshot from 2024-02-03 11-05-22](https://github.com/thorvg/thorvg/assets/4047289/dd41bd29-da06-4156-a75d-908892dd6bb3)

**Comment by hermet** [comment]
@lostminds @capnm Thansk for reporting, hopefully this will resolve that issue. https://github.com/thorvg/thorvg/pull/1957

**Comment by hermet** [comment]
The fix will be applied in v0.12.5
